### PR TITLE
Security fix

### PR DESCRIPTION
**Pull request by belosh59**

### CODE REVIEW

File: goods-partner-backend/src/main/java/com/goodspartner/configuration/security/SecurityConfig.java [hunk 1]
```diff
@@ -23,28 +23,41 @@
 
 @Configuration
 @EnableWebSecurity
-@EnableGlobalMethodSecurity(prePostEnabled = true)
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/configuration/security/SecurityConfig.java:26]: Why we remove this?

File: goods-partner-backend/src/main/java/com/goodspartner/configuration/security/SecurityConfig.java [hunk 2]
```diff
@@ -57,4 +70,9 @@ CorsConfigurationSource corsConfigurationSource() {
         source.registerCorsConfiguration("/**", configuration);
         return source;
     }
+
+    @ConditionalOnProperty(prefix = "goodspartner.security", name = "enabled", havingValue = "true", matchIfMissing = true)
+    @EnableGlobalMethodSecurity(prePostEnabled = true)
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/configuration/security/SecurityConfig.java:75]: Why we need such structure via static class?

File: goods-partner-backend/src/main/java/com/goodspartner/exception/UserNotFoundException.java [hunk 3]
```diff
@@ -2,19 +2,19 @@
...
 public class UserNotFoundException extends RuntimeException {
 
-    private static final String USER_NOT_FOUND_BY_EMAIL_MESSAGE = "User not found with email: %s";
+    private static final String USER_NOT_FOUND_BY_LOGIN_MESSAGE = "User not found with login: %s";
     private static final String USER_NOT_FOUND__BY_ID_MESSAGE = "User not found with id: %d";
     private static final String USER_NOT_FOUND_BY_ID_ROUTE_ID = "User not found for route ID: %s";
 
-    private static final String USER_NOT_FOUND_BY_EMAIL_MESSAGE_UKR = "Користувача з email: %s не існує";
-    private static final String USER_NOT_FOUND__BY_ID_MESSAGE_UKR = "Користувача з id: %d не існує";
+    private static final String USER_NOT_FOUND_BY_EMAIL_LOGIN_UKR = "Користувача з login: %s не існує";
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/exception/UserNotFoundException.java:9]: Користувача з login: 

login????

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultUserService.java [hunk 4]
```diff
@@ -86,6 +86,7 @@ private String getUsernameFromAuthenticationContext() {
         if (authentication.isAuthenticated()) {
             UserDetails principal = (UserDetails) authentication.getPrincipal();
             return principal.getUsername();
+//            return authentication.getPrincipal().toString();
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultUserService.java:89]: remove it?

**Review by Rooman — DISMISSED** [review]

File: goods-partner-backend/src/main/resources/application-prod.yml [hunk 5]
```diff
@@ -3,16 +3,16 @@ spring:
...
+  security:
+    oauth2:
+      client:
+        registration:
+          google:
+            client-id: ${CLIENT_ID}
+            client-secret: ${CLIENT_SECRET}
+            scope:
+              - email
+              - profile
```
Comment (belosh59) [inline goods-partner-backend/src/main/resources/application-prod.yml:15]: OAuth2 we keep here?

**Review by Rooman — APPROVED** [review]

**Pull request merged**